### PR TITLE
fix(html_report): add if statement

### PR DESCRIPTION
**Pull request by ashrivastava-qa**
## Because

- The reports were not generating for parallel machines that had failed tests runs.

## This pull request

- Added the `when` parameter to config.yaml file for the 'rename reports' step
- Added condition to rename files when the reports otherwise provide an apt message 
- Added if condition for merging the reports when the report exist otherwise provide an apt message

## Issue that this pull request solves

Closes: FXA-9885 FXA-9979

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [ ] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

### CODE REVIEW

File: .circleci/config.yml
Comment (Trinaa) [inline .circleci/config.yml:350]: Would it be possible to set the report name when we execute Playwright using `PLAYWRIGHT_BLOB_OUTPUT_NAME` or one of the other env vars listed [here](https://playwright.dev/docs/test-reporters#blob-reporter) instead, avoiding the need for this step.

**Comment by Trinaa** [comment]
Not sure if these things should be in scope in this fix, but we should probably consider the following:

1. The HTML report doesn't get generated in the case where the functional tests fail:
image

2. [CircleCI documentation](https://circleci.com/docs/artifacts/#artifacts-and-self-hosted-runner) seems to indicate that there are 'cost implications' to storing artifacts. I wonder if we should omit artifacting the blobs and just upload the report? Or at least stop uploading it three times ; In `Functional Tests - Playwright` we upload the original blobs and the renamed blobs and in `Merge Playwright Reports` we upload the merged `blob-report/report.zip`, but it's all the same data. The traces are also getting duplicated.
image
image

cc: @pdehaan

File: .circleci/config.yml
Comment (ashrivastava-qa) [reply .circleci/config.yml:350]: Yes, I tried that [here](https://github.com/mozilla/fxa/pull/17089), but it wasn't working for our `config.yaml` file somehow. I tried a number of things which accounts for the 55 commits in that PR :D 
And finally I had to resort to adding the 'renaming' step so that the `report.zip` file doesn't create a problem while merging, which it was (because of the same name).

**Comment by ashrivastava-qa** [comment]
I am thinking we should create a separate ticket for the things you have mentioned @Trinaa and merge this change so that it unblocks the PRs and the use of 're-run from failed' option. What do you think @pdehaan ?

**Comment by ashrivastava-qa** [comment]
@Trinaa Yes I can see and try to minimize the upload. But just 'cost implication' sounds vague to me, due you have any insight on how much this cost is? If I can get an idea I might try different things to compare the cost.

**Comment by ashrivastava-qa** [comment]
@Trinaa and @pdehaan I created a [separate issue](https://mozilla-hub.atlassian.net/browse/FXA-9979) to address both the above mentioned items Kat pointed out. I think if this looks okay we can merge this for now to unblock the team and I can work on the other ticket separately.

File: .circleci/config.yml
Comment (pdehaan) [inline .circleci/config.yml:689]: What is this line doing?
I guess specifically the `1> /dev/null 2>&1`. Is it just counting the number of ZIP files?

Comment (ashrivastava-qa) [reply .circleci/config.yml:689]: The expression `1> /dev/null 2>&1` is a shell command syntax used to handle input and output redirection. It means:
`1> /dev/null`: Redirect the stdout to /dev/null (discard it).
`2>&1`: Redirect stderr to wherever stdout is currently being redirected (which, in this case, is /dev/null).
In essence, this combination is used to suppress all output from a command, both `stdout` and `stderr`.
This command is useful in scripts where you want to check for the presence of files or execute a command silently without producing any output or error messages. Here it checks if any .`zip` files exist in the `artifacts/blob-report` directory. If files exist, the subsequent merge command is executed. If no files exist, a message is printed and the merge command is skipped. This prevents errors when trying to merge reports if no report files are present. Though it is a temp solution for now.

**Comment by ashrivastava-qa** [comment]
@Trinaa @pdehaan 
1. I was able to fix the script so that report-[0-7].zip generates even for the failed test machine.
2. For minimizing the artifacts upload, we are uploading the report.zip file thrice (twice in Functional-Test Playwright job and once in Merge Reports job). However the problem I am encountering when trying to minimize this is:
a. If I remove the `blob report directory` path from `Playwright.config.ts file`, which is duplicating the `reports.zip` file in all the machines, no `reports.zip` file generates at all. Couldn't understand why as the Playwright docs suggests its either the path needs to be setup or the Environment variable but somehow we have to use `path` with or without Env variables.

b. The second upload is just renaming (not copying) the report.zip file to report-[0-7].zip file and store in artifacts/blob-report so that it can be merged.


c. The last upload is the merged file again in `artifacts/blob-report` for `Merge Reports job` to be able to use it for the HTML reporting. And this is required can’t not skip this upload.

d. Tried to remove the 'rename reports' step altogether and use `PLAYWRIGHT_BLOB_OUTPUT_FILE="report-$(CIRCLE_NODE_INDEX).zip` in `config.yaml` and `playwright.config.ts` using the output filename but that didn't work either.


So, IMO the last 2 steps are required but the first step could be removed which is the cause of duplication but somehow removing it doesn't generate the reports file at all. And the CircleCI doc states  “Retaining an artifact for a long period of time will have storage cost implications, therefore, it is best to determine why you are retaining artifacts” and we only retain the data for 90days so I am guessing it won’t be too costly and I talked to Ben as well and he mentioned in Nimbus/experimenter he has seen the cost to be minimal for artifacts upload as close to .0001 cent or something. 

So what are your thoughts, should we keep the extra upload happening in step 1 or if you have any insights into how to remove it but still make the blob reporter work I'll be happy to try it out.

**Review by vbudhram — APPROVED** [review]
@ashrivastava-qa Makes sense to me 👍🏽

**Pull request merged**
